include/sparrow/layout/variable_size_binary_view_array.hpp

```diff
@@ -0,0 +1,369 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstddef>
+#include <ranges>
+
+#include "sparrow/arrow_interface/arrow_array.hpp"
+#include "sparrow/arrow_interface/arrow_schema.hpp"
+#include "sparrow/utils/functor_index_iterator.hpp"
+#include "sparrow/arrow_interface/arrow_schema.hpp"
+#include "sparrow/arrow_array_schema_proxy.hpp"
+#include "sparrow/layout/array_bitmap_base.hpp"
+#include "sparrow/layout/layout_utils.hpp"
+#include "sparrow/utils/iterator.hpp"
+#include "sparrow/utils/nullable.hpp"
+#include "sparrow/layout/array_access.hpp"
+#include "sparrow/buffer/dynamic_bitset.hpp"
+#include "sparrow/buffer/u8_buffer.hpp"
+#include "sparrow/utils/ranges.hpp"
+
+namespace sparrow
+{
+
+
+    template <class T>
+    class variable_size_binary_view_array_impl;
+    using string_view_array = variable_size_binary_view_array_impl<std::string_view>;
+    using binary_view_array = variable_size_binary_view_array_impl<std::span<const std::byte>>;
+
+    namespace detail
+    {
+        template<class T>
+        struct get_data_type_from_array;
+
+        template<>
+        struct get_data_type_from_array<sparrow::string_view_array>
+        {
+            constexpr static sparrow::data_type get()
+            {
+                return sparrow::data_type::STRING_VIEW;
+            }
+        };
+        template<>
+        struct get_data_type_from_array<sparrow::binary_view_array>
+        {
+            constexpr static sparrow::data_type get()
+            {
+                return sparrow::data_type::BINARY_VIEW;
+            }
+        };
+    }
+
+    template <class T>
+    struct array_inner_types<variable_size_binary_view_array_impl<T>> : array_inner_types_base
+    {
+        using array_type = variable_size_binary_view_array_impl<T>;
+        using inner_value_type = T;
+        using inner_reference = T;
+        using inner_const_reference = inner_reference;
+    
+        using value_iterator = functor_index_iterator<detail::layout_value_functor<array_type, inner_value_type>>;
+        using const_value_iterator = functor_index_iterator<
+            detail::layout_value_functor<const array_type, inner_reference>>;
+        using iterator_tag = std::random_access_iterator_tag;
+    };
+
+    template <class T>
+    struct is_variable_size_binary_view_array_impl : std::false_type
+    {
+    };
+
+    template <class T>
+    struct is_variable_size_binary_view_array_impl<variable_size_binary_view_array_impl<T>> : std::true_type
+    {
+    };
+
+    /**
+     * Checks whether T is a variable_size_binary_view_array_impl type.
+     */
+    template <class T>
+    constexpr bool is_variable_size_binary_view_array_impl_v = is_variable_size_binary_view_array_impl<T>::value;
+
+    template <class T>
+    class variable_size_binary_view_array_impl final : public mutable_array_bitmap_base<variable_size_binary_view_array_impl<T>>
```

Comment:
Why not `variable_size_binary_view_array` instead? (I am doing the same in variable_size_binary_array btw).

Reply:
the users should not use use class with templates because only where few instantations are allowed

Reply:
Indeed but they will use the "is_variable_size_binary_view_array_impl", so this is not consistent. Or we provide `is_string_view_array` and  `is_binary_view_array` that compare to the instantiations instead of having a generic comparison.